api/fs.go

```diff
@@ -315,8 +320,11 @@ func (a *AllocFS) Logs(alloc *Allocation, follow bool, task, logType, origin str
 			// Decode the next frame
 			var frame StreamFrame
 			if err := dec.Decode(&frame); err != nil {
-				errCh <- err
-				close(frames)
+				if err == io.EOF {
```

Comment:
Put my review comment in the wrong place so repasting it here: 

Shouldn't we also close the channel when the error is ErrClosedPipe?

Reply:
Good catch! I think due to how we _currently_ Close the pipe that can't happen, but there's no reason not to be defensive.